src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Split.cs

```diff
@@ -87,7 +87,36 @@ private static string[] Split(Regex regex, string input, int count, int startat)
             count--;
             (List<string> results, int prevat, string input, int count) state = (results: new List<string>(), prevat: 0, input, count);
 
-            if (!regex.RightToLeft)
+            if (regex._srm is not null)
```

Comment:
Subsequently we should spend a little time seeing how we might better abstract away the implementations to not need these special cases sprinkled throughout the top-level APIs.  What you have here is good for now, though.

Reply:
I was thinking the same, but found it too difficult myself to do a deeper integration through the Run method of Regex. It really matters mostly for Replace and Split where things could be more unified I guess, they have similar top-level loops, but there are also some differences (mainly concerning wrt captures), and the new case is integrated similarly.

Reply:
>> we should spend a little time seeing how we might better abstract away the implementations to not need these special cases sprinkled throughout the top-level APIs

> I was thinking the same, but found it too difficult myself to do a deeper integration through the Run method of Regex

I'm working on this.